### PR TITLE
Restore online buildpack functionality

### LINKED ISSUE #102: Online buildpack failing with error "does not support multi-buildpack apps"
## Summary
Using the online buildpack fails with the error:
```
 ERR Error: one of the buildpacks chosen to supply dependencies does not support multi-buildpack apps
```

## Steps to Reproduce
1. Clone this project and change to the [`tests/integration/apps/ruby`](https://github.com/cyberark/cloudfoundry-conjur-buildpack/tree/master/tests/integration/apps/ruby) directory.

2. Add a file `manifest.yml` which contains
    ```
    ---
    applications:
    - name: ruby-app
      env:
        CONJUR_BUILDPACK_BYPASS_SERVICE_CHECK: true
      buildpacks:
      - https://github.com/cyberark/cloudfoundry-conjur-buildpack
      - ruby_buildpack
    ```

3. `cf push` from the Ruby app directory

## Expected Results
The app should successfully push and when you visit the app in the browser you should see:
```
      <h1>Visit us @ www.conjur.org!</h1>

      <h3>Space-wide Secrets</h3>
      <p>Database Username: space_username</p>
      <p>Database Password: space_password</p>
```

## Actual Results (including error logs, if applicable)
The app fails to be staged on `cf push`:
```
$ cf push
Pushing from manifest to org geri-org / space geri-space as admin...
Using manifest file /Users/gjennings/Documents/repos/pcf/buildpack/tests/integration/apps/ruby/manifest.yml
Getting app info...
Creating app with these attributes...
+ name:         ruby-app
  path:         /Users/gjennings/Documents/repos/pcf/buildpack/tests/integration/apps/ruby
  buildpacks:
+   https://github.com/cyberark/cloudfoundry-conjur-buildpack
+   ruby_buildpack
  env:
+   CONJUR_BUILDPACK_BYPASS_SERVICE_CHECK
  routes:
+   ruby-app.apps.pcf.itd.conjur.net

Creating app ruby-app...
Mapping routes...
Comparing local files to remote cache...
Packaging files to upload...
Uploading files...
 2.19 KiB / 2.19 KiB [==============================================] 100.00% 1s

Waiting for API to complete processing files...

Staging app and tracing logs...
   Downloading ruby_buildpack...
   Downloaded ruby_buildpack
   Cell f285e943-6329-44c3-a703-4c6cf25cf19b creating container for instance cff4f7e6-9269-4a54-b07e-5db1278d7dd0
   Cell f285e943-6329-44c3-a703-4c6cf25cf19b successfully created container for instance cff4f7e6-9269-4a54-b07e-5db1278d7dd0
   Downloading app package...
   Downloaded app package (2.2K)
Error staging application: Staging error: staging failed
FAILED
```

When you look at the app logs, you see:
```
ruby$ cf logs ruby-app
Retrieving logs for app ruby-app in org geri-org / space geri-space as admin...

   2020-12-21T10:50:45.14-0500 [STG/0] OUT Downloaded app package (2.2K)
   2020-12-21T10:50:45.68-0500 [STG/0] ERR Error: one of the buildpacks chosen to supply dependencies does not support multi-buildpack apps
   2020-12-21T10:50:45.70-0500 [STG/0] OUT Exit status 225
   2020-12-21T10:50:45.83-0500 [STG/0] OUT Cell f285e943-6329-44c3-a703-4c6cf25cf19b stopping instance cff4f7e6-9269-4a54-b07e-5db1278d7dd0
   2020-12-21T10:50:45.83-0500 [STG/0] OUT Cell f285e943-6329-44c3-a703-4c6cf25cf19b destroying container for instance cff4f7e6-9269-4a54-b07e-5db1278d7dd0
   2020-12-21T10:50:45.93-0500 [API/0] ERR Failed to stage build: staging failed
   2020-12-21T10:50:46.12-0500 [STG/0] OUT Cell f285e943-6329-44c3-a703-4c6cf25cf19b successfully destroyed container for instance cff4f7e6-9269-4a54-b07e-5db1278d7dd0
```

In preparation for #101 there is a copy of the latest release in the `latest` branch; if you update the manifest to refer to `https://github.com/cyberark/cloudfoundry-conjur-buildpack#latest`, it works as expected.

## Reproducible
   * [x] Always 
   * [ ] Sometimes
   * [ ] Non-Reproducible
   
## Version/Tag number
The online buildpack uses the latest code on master.

## Environment setup
This test case was run in TAS 2.10, but should function the same on other earlier TAS versions.

### PR DESCRIPTION
**Pull request by izgeri**
### What does this PR do?
- Adds test flows for the online buildpack in a remote TAS foundation.
- Reorganizes the project to work again as either an online or offline buildpack - see the commit message for more detailed information. Basically this involved restoring the buildpack scripts to the `bin/` directory and making sure `buildpack-packager` could still package the offline buildpack correctly.
- Bumps the minimum Go version to 1.14.

I don't include a change log message with this change because it fixes an issue in the last unreleased change - when we next release this project, it's irrelevant to the end user that we broke something and then fixed it between releases.

**Note:** We still recommend, as will be implemented in #101, that users move to using the `latest` branch for the online buildpack to avoid issues with the `master` branch having development updates that break online functionality.

### What ticket does this PR close?
Resolves #102 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: ci/test_integration [hunk 1]
```diff
@@ -17,7 +17,9 @@ setup_env
 build_test_images
 start_conjur
 
-# If we're running in Jenkins, skip this step
+# If we're running in Jenkins, skip this step since it's already been
```
Comment (diverdane) [inline ci/test_integration:20]: Nice to have this explanation!

File: package.sh [hunk 2]
```diff
@@ -28,11 +28,14 @@ docker run --rm \
   -v $(pwd):/cyberark \
   packager \
   /bin/bash -c """
-  # Create pkg folder that is the repository of the final artefacts
+  # Create pkg folder that is the repository of the final artifacts
   mkdir /pkg
-  # Copy the final artefacts to pkg
+  mkdir /pkg/bin
```
Comment (diverdane) [inline package.sh:33]: We could skip the preceding `mkdir /pkg` and do:
```
    mkdir -p /pkg/bin
    mkdir -p /pkg/buildpack
```

File: tests/integration/features/integration.feature [hunk 3]
```diff
@@ -15,23 +15,35 @@ Feature: Integrations Tests for PCF 2.4
...
+      When I push a "java" app with the "offline" buildpack
       Then the secrets.yml values are available in the app
 
 #    # The online buildpack tests are only valid if the latest commits
 #    # are push to the Github remote branch.
-#    Scenario: Java online buildpack integration
-#      When I push a Java app with the online buildpack
-#      Then the secrets.yml values are available in the app
+    Scenario: Python online buildpack integration
+      When I push a "python" app with the "online" buildpack
```
Comment (diverdane) [inline tests/integration/features/integration.feature:36]: Online tests, NOICE!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -60,18 +60,21 @@ This will automatically update go.mod and go.sum.
...
-  Available versions and hashes can be found [here][buildpacks],
-  or see the manifest for the [official Go Buildpack][go-buildpacks].
+When upgrading the version of Go for `conjur-env`, the value needs to be updated
+in a few places:
+
+* Update the base image in `.buildpack/conjur-env/Dockerfile
+* Update the Go version in `./buildpack/conjur-env/go.mod`
+* Update the version and file hashes in `manifest.yml` - available versions and
+  hashes can be found [here][buildpacks], or see the manifest for the
+  [official Go Buildpack][go-buildpacks]. (This is for the offline version of
```
Comment (diverdane) [inline CONTRIBUTING.md:70]: This was broken before this change, but `go-buildpacks` link should be `go-buildpack`.

File: bin/supply [hunk 5]
```diff
@@ -5,19 +5,19 @@
...
-CACHE_DIR=$2
+# CACHE_DIR=$2
 
 # App dependencies are stored in $DEPS_DIR/$INDEX_DIR.
 DEPS_DIR=$3
 INDEX_DIR=$4
 
-BIN_DIR=$(cd $(dirname $0); pwd)
-BUILDPACK_DIR=$(dirname $BIN_DIR)
+BIN_DIR=$(cd $(dirname "$0"); pwd)
```
Comment (diverdane) [inline bin/supply:14]: I'm seeing a CodeClimate warning on GitHub, I think it wants nested double quotes for this and the next line:
```
BIN_DIR="$(cd $(dirname "$0"); pwd)"
BUILDPACK_DIR="$(dirname "${BIN_DIR}")"/buildpack
```

File: bin/supply [hunk 6]
```diff
@@ -28,46 +28,72 @@ if  [ -f $BUILD_DIR/BOOT-INF/classes/secrets.yml ]; then
...
+else
+  echo " - need to compile conjur-env"
+  echo "[cyberark-conjur-buildpack]: install go to build conjur-env binary"
+  source $BUILDPACK_DIR/lib/install_go.sh
+
+  pushd "${DEPS_DIR}/${INDEX_DIR}"
+    cd "${BUILDPACK_DIR}/conjur-env"
+
+    echo "[cyberark-conjur-buildpack]: download the conjur-env binary dependencies"
+    $GoInstallDir/bin/go mod download
```
Comment (diverdane) [inline bin/supply:72]: Should probably have double-quotes around `$GoInstallDir`.

File: buildpack/lib/install_go.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Borrowing this script from the Node.js Buildpack.
+# https://raw.githubusercontent.com/cloudfoundry/nodejs-buildpack/master/scripts/install_go.sh
+
+set -euo pipefail
+
+GO_VERSION="1.14"
+
+if [ $CF_STACK == "cflinuxfs3" ]; then
```
Comment (diverdane) [inline buildpack/lib/install_go.sh:10]: Add double quotes around `$CF_STACK`.

**Review by diverdane — COMMENTED** [review]
Looks good!! Just a few editorial nits.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -60,18 +60,21 @@ This will automatically update go.mod and go.sum.
...
-
-* **Offline build:** Update the base image version in `.buildpack/conjur-env/Dockerfile`
-
-* **Online build:** Update the version and file hashes in `manifest.yml`.
-  Available versions and hashes can be found [here][buildpacks],
-  or see the manifest for the [official Go Buildpack][go-buildpacks].
+When upgrading the version of Go for `conjur-env`, the value needs to be updated
+in a few places:
+
+* Update the base image in `.buildpack/conjur-env/Dockerfile
```
[the remark this answers is not in the document]
Comment (izgeri) [reply CONTRIBUTING.md:66]: perils of a blue terminal window and a dark red backtick

File: package.sh (hunk 2 above)
Comment (izgeri) [reply package.sh:33]: yeah, you're right

File: tests/integration/features/integration.feature (hunk 3 above)
Comment (izgeri) [reply tests/integration/features/integration.feature:36]: YAS

File: buildpack/lib/install_go.sh (hunk 7 above)
Comment (izgeri) [reply buildpack/lib/install_go.sh:10]: I have done this 2x already ... the case of the Great Disappearing CF Stack Double Quotes

Comment (izgeri) [reply buildpack/lib/install_go.sh:10]: HA I just lost them a 3rd time and I finally figured out why ... fixing

**Review by diverdane — APPROVED** [review]
LGTM!!!

**Pull request merged**